### PR TITLE
Developed Dashboard Hardware Metrics Feature

### PR DESCRIPTION
**Pull request by manoj22-k**
- Implemented HardwareHealth functional component in UI and wrote unit testcases
- Implemented get_harware_health api in python BE and wrote unit testcases
- Displayed number of criticals, warnings and nominals for each and overall metrics
image
- Displayed values of similar types of metrics
image

### CODE REVIEW

**Comment by mjlee34** [comment]
Which hardware metrics are displayed? Could you let me know the list?
Additionally, for better understanding, please upload an example screenshot.

Thanks in advance.

**Comment by manoj22-k** [comment]
Hi @mjlee34, 
I have updated 1st comment and attached screenshots there. 
The Hardware metrics list which is being used are following:
    bmc
	ipmi_fan_speed_state
	ipmi_fan_speed_rpm
	ipmi_power_state
	ipmi_power_watts
	ipmi_sensor_state
	ipmi_sensor_value
	ipmi_voltage_state
	ipmi_voltage_volts
	ipmi_temperature_state
	ipmi_temperature_celsius
	ipmi_chassis_power_state
    Device
	temperature
	warning_temperature_time
	critical_temperature_time
	available_spare
	available_spare_threshold
Thanks for comment

**Review by vishal-shakya-ssir — APPROVED** [review]
LGTM

**Review by aswinssir — APPROVED** [review]
LGTM

File: src/mtool/api/rest/app.py [hunk 1]
```diff
@@ -2086,6 +2075,30 @@ def check_telemetry(current_user):
     except Exception as e:
         return make_response('Prometheus DB is not running'+str(e), 500)
 
+@app.route('/api/v1/perf/all', methods=['GET'])
+def get_current_iops():
+    try:
+        received_telemetry = connection_factory.get_telemetery_url()
+        ip = received_telemetry[0]
```
[the remark this answers is not in the document]
Comment (manoj22-k) [reply src/mtool/api/rest/app.py:2084]: Added check for empty list. Thanks for review.

File: src/mtool/api/rest/app.py [hunk 2]
```diff
@@ -2086,6 +2075,30 @@ def check_telemetry(current_user):
...
+    except Exception as e:
+        return make_response('Could not get performance metrics'+str(e), 500)
+
+@app.route('/api/v1/get_hardware_health', methods=['GET'])
+@token_required
+def get_hardware_health(current_user):
+    try:
+        received_telemetry = connection_factory.get_telemetery_url()
+        ip = received_telemetry[0]
+        port= received_telemetry[1]
```
[the remark this answers is not in the document]
Comment (manoj22-k) [reply src/mtool/api/rest/app.py:2099]: added check.

**Pull request merged**